lib/nats.js

```diff
@@ -1166,9 +1166,16 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
         opt_callback = subject;
         subject = undefined;
     }
-    if (!msg) {
-        msg = EMPTY;
+
+    if (!this.options.json) {
+        if (!msg) {
```

Comment:
The test here will need to be more careful - as the provided values could be `false`, `""` (empty string), `null`, `undefined`, and only in the case of undefined should this value be touched to be `null`